### PR TITLE
Integrate date filter

### LINKED ISSUE #38: Integrate DateFilter with the backend
# Context

* The DateFilter modal is ready, but not functional. We need to make it functional, so users can filter their appointments in the schedule page.
# Tech planning

*tbd

# Acceptance criteria

* Users can filter their appointments in the schedule page.

### PR DESCRIPTION
**Pull request by IagoGoncalves22**
Resolves #38 

## Problem

We don't have DateFilter integrated yet with teh backend.

## Changes proposed in this PR

1. Improved the SchedulePage.tsx with useSearchParams and weekfilter.
2. Improved the SchedulePage.tsx with endOfWeek from date-fns.
3. Improved the SchedulePage.tsx with format and startOfWeek from date-fns/fp.
4. Implemented EmptyState in the file SchedulePage.tsx.
5. Created the file EmptyStates.tsx and implemented add appointment modal in the file.
6. Improved translate of week filter and empty state.

### CODE REVIEW

File: public/locales/pt.json [hunk 1]
```diff
@@ -77,7 +77,10 @@
         "schedule": "Agenda",
         "customer": "Cliente",
         "time": "Horário",
+        "thisWeek": "Semana",
```
Comment (galexandrade) [inline public/locales/pt.json:80]: Não seria `Esta semana`?

File: src/components/EditAppointmentModal.tsx [hunk 2]
```diff
@@ -37,10 +37,10 @@ const EditAppointmentModal = ({
         day: Yup.date().required(t('scheduleRow.schemaDay')),
         time: Yup.string().required(t('scheduleRow.schemaTime'))
     });
-
+    console.log(appointment.day);
```
Comment (galexandrade) [inline src/components/EditAppointmentModal.tsx:40]: I think you can remove this.

File: package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "version": "0.1.0",
     "private": true,
     "dependencies": {
-        "@7shifts/sous-chef": "^3.3.5",
+        "@7shifts/sous-chef": "^3.19.0",
```
Comment (galexandrade) [inline package.json:6]: It seems something wrong happened with the Card after we updated the Sous CHef version 🤔 

image


But you can quickly fix it by adding this CSS code to the `src/pages/LoginPage.scss` file:
image

File: src/pages/SchedulePage.tsx [hunk 4]
```diff
@@ -75,6 +119,13 @@ const SchedulePage = () => {
                     {t('schedulePage.addSchedule')}
                 </Button>
             </Inline>
+
+            {!isLoading && days.length === 0 ? (
+                <EmptyStates onScheduleAdded={() => loadCustomers()} />
+            ) : null}
+
+            {isLoading && <Spinner block />}
+
```
Comment (galexandrade) [inline src/pages/SchedulePage.tsx:128]: I think we could give 36px of space between the tables to it looks better.
You can wrap the `{days.map}` in a `Stack` and it should do the trick (don't for get to close it at the end)

image

Also, we could give 12px space from the header to the table. You can also use `Stack` for that:
image

**Review by galexandrade — COMMENTED** [review]
It is looking good. Just some small details.

File: public/locales/pt.json (hunk 1 above)
Comment (IagoGoncalves22) [reply public/locales/pt.json:80]: Right. It was fixed.

File: package.json (hunk 3 above)
Comment (IagoGoncalves22) [reply package.json:6]: Great. It was solved.

File: src/pages/SchedulePage.tsx (hunk 4 above)
Comment (IagoGoncalves22) [reply src/pages/SchedulePage.tsx:128]: Alright it was done.

**Review by galexandrade — APPROVED** [review]
It looks nice. Great job 🎉

**Pull request merged**